ravenframework/Models/EnsembleModel.py

```diff
@@ -136,7 +136,7 @@ def localInputAndChecks(self,xmlNode):
           self.raiseAnError(IOError, "Input XML node for Model" + modelName +" has not been inputted!")
         if len(self.modelsInputDictionary[modelName].values()) > allowedEntriesLen:
           self.raiseAnError(IOError, "TargetEvaluation, Input and metadataToTransfer XML blocks are the only XML sub-blocks allowed!")
-        if child.attrib['type'].strip() == "Code":
+        if child.attrib['type'].strip() in ["Code", 'HybridModel', 'LogicalModel']:
           self.createWorkingDir = True
```

Comment:
It is ugly and in addition this creates a sub-working directory even if, for example, the Logical/hybrid models do not use a ```Code```. In case of HybridModel/LogicalModel using only ExternalModels/ROMs, the subdirectory is created but will stay empty. Not very elegant. @joshua-cogliati-inl @wangcj05 any ideas on how to improve this?

Reply:
@wangcj05 @joshua-cogliati-inl any ideas for this? I cannot find a better solution at this stage

Reply:
Can we have a class attribute to indicate if there is a code associated with the Model? For example, In Hybrid Model/Logical/Ensemble Model, we define a `self._isCodeAvail`, and assign it to true when we detect a code in the Model. @alfoa

Reply:
Hm, I am trying to fully understand why you need to create the directory?   Just to check, it is needed if the Logical/hybrid models use a Code?

Reply:
yes that's why. If there is a Code in the underlying Logical/Hybrid model (contained in the ensemble model) the subfolder is required.

Reply:
@wangcj05 @joshua-cogliati-inl can you tell me how exactly you would like that flag to be coded? (I prefer not to take a code design decision (that might be needed to be modified) on my own)?

Reply:
@wangcj05 @joshua-cogliati-inl  FY: if you can send feedbacks within tomorrow I can try to address them before leaving on Friday. Otherwise it will need to wait till September.